"Add User" documentation (#150)
astronomer/cloud-docs: docs/add-users.md

---
sidebar_label: 'Add Users to a Workspace'
title: 'Add Users to a Workspace'

> id: add-users

Reviewer: I also lean towards "user" vs. "users" since they can't add more than 1 user at the same time right now and it's one action (even tho you can do it multiple times)

---

## Overview

> As a Workspace Admin, you can add users with Astronomer Cloud accounts to your Workspace. From here, you can grant them a role with permissions for specific actions across the Workspace.

Reviewer: ```suggestion
As a Workspace Admin, you can add members of your team to your Astronomer Cloud Workspace. From here, you can grant them a variety of user roles with a defined set of permissions for specific actions across the Workspace and the Deployments within it.
```

Reviewer: This may not be ideal wording but I'm thinking 1) something about permissions applying to Deployments and 2) something about the "team" part of "inviting your team to Astronomer" and then 3) feel like we don't need to say "users with Astronomer Cloud accounts"

Author: Do we want to start using "Team" in the non-RBAC context? I think it's okay now, but that word will soon be overloaded.

> As a Workspace Admin, you can invite users with Astronomer accounts to your Workspace. From here, you can grant them a role with permissions to complete certain actions across the Workspace.

Reviewer 2: Let's be more specific.

Astronomer Cloud accounts vs just Astronomer accounts.

Reviewer 2: Also let's say add vs invite. We need to build out the invite feature, which would be at the Org level.

This guide provides steps for adding a user to an Astronomer Workspace.

## Prerequisites

> To add users to a Workspace, you must have:
> 
> - An [Astronomer installation](install-aws).
> - Workspace Admin permissions.

Reviewer: ```suggestion
To add a user to a Workspace, you must have:

- An [Organization on Astronomer Cloud](install-aws).
- Workspace Admin permissions.
```

Let's get away from "installation" and just to Org

## Step 1: Invite the User to Astronomer

> If the user you want to add doesn't already have an Astronomer account, tell them to go to `cloud.astronomer.io` and sign up. When they successfully create an account, they will see the following screen:
> 
> <div class="text--center">
>   <img src="/img/docs/welcome-user.png" alt="Successful account creation screen" />
> </div>
> 
> Make note of the user's account email for the next step.

Reviewer: ```suggestion
If the user you want to add doesn't already have an account on Astronomer Cloud, tell them to go to `cloud.astronomer.io` and sign up. Once they create an account, they will see the following screen:

<div class="text--center">
  <img src="/img/docs/welcome-user.png" alt="Successful account creation screen" />
</div>

Make note of the user's account email for the next step.
```

Reviewer: Should we add something about authentication method?

## Step 2: Add the User

> To add a user with an Astronomer account to your Workspace:

Reviewer: ```suggestion
To add a user with an Astronomer Cloud account to your Workspace:
```

1. In the Astronomer UI, open your Workspace and go to the **Access** tab.
2. Click **Add member**:

    <div class="text--center">
      <img src="/img/docs/add-user.png" alt="Add member button in the access menu" />
    </div>

3. Enter the user's email from Step 1.
4. Set a Workspace role for the user. The available roles are:

>     - **Workspace Viewer**: Has permission to view anything in the Workspace, including users, Deployments, and pipeline code
>     - **Workspace Editor**: Has Viewer permissions, plus permission to [deploy code](deploy-code) and [configure Deployments](configure-deployment)
>     - **Workspace Admin**: Has Editor permissions, plus permission to add users, remove users, and assign user roles

Reviewer: ```suggestion
    - **Workspace Viewer**: Has permission to view anything in the Workspace, including users, Deployments, and pipeline code in the Airflow UI
    - **Workspace Editor**: Has Viewer permissions, plus permission to [deploy code](deploy-code) and [configure Deployments](configure-deployment)
    - **Workspace Admin**: Has Editor permissions, plus permission to add users, remove users, and assign user roles
```

Reviewer: Where do we mean "pipeline code"?

Author: Should be "DAG code", but I'm not sure how we're terming that currently - one for the style guide?

5. Click **Add member**.

> Once you successfully add the user, their information will appear in the **Access** tab as a new entry in the **Members** table.

Reviewer: ```suggestion
Once you add the user, their information will appear in the **Access** tab as a new entry in the **Members** table.
```
